### PR TITLE
add script for preseal a public key to tpm

### PR DESCRIPTION
**Pull request by jpbland1**
add more detailed instructions on how to prepare a TPM keystore build and how to use the preseal script

tested in simulator for first time boot with unseal

### CODE REVIEW

File: tools/preseal/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+all: preseal
+
+preseal:
+	$(Q)$(CC) -o $@ $@.c -lwolftpm
+
+debug:
+	$(Q)$(CC) -o preseal preseal.c -g -lwolftpm -lwolfssl
+
+clean:
```
Comment (dgarske) [inline tools/preseal/Makefile:11]: Looks like you added the binary. It should not be tracked and it should be in the .gitignore.

File: tools/preseal/preseal.c [hunk 2]
```diff
@@ -0,0 +1,195 @@
+#include <wolfssl/options.h>
```
Comment (dgarske) [inline tools/preseal/preseal.c:1]: Please add our standard license header

File: tools/preseal/preseal.c [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+#include <wolfssl/wolfcrypt/settings.h>
```
Comment (dgarske) [inline tools/preseal/preseal.c:2]: Does this tool really work against a physically different TPM? Or does it need to be run using the same TPM??

**Review by dgarske — CHANGES_REQUESTED** [review]

File: tools/preseal/preseal.c (hunk 3 above)
Comment (jpbland1) [reply tools/preseal/preseal.c:2]: It needs to run on the same TPM since it needs to seal the public key to NV

**Review by dgarske — APPROVED** [review]

**Pull request merged**